### PR TITLE
Group functions by extensions in contract explorer

### PR DESCRIPTION
**Pull request by joaquim-verges**

### CODE REVIEW

File: components/contract-functions/contract-function.tsx [hunk 1]
```diff
@@ -144,27 +152,78 @@ interface ContractFunctionsPanelProps {
...
-  const viewFunctions: AbiFunction[] = useMemo(() => {
-    return fnsOrEvents.filter(
-      (fn) =>
-        (fn as AbiFunction).stateMutability === "pure" ||
-        (fn as AbiFunction).stateMutability === "view",
-    ) as AbiFunction[];
-  }, [fnsOrEvents]);
+  const writeFunctions: ExtensionFunctions[] = useMemo(() => {
+    return functionsWithExtension
+      .map((e) => {
```
Comment (assimovt) [inline components/contract-functions/contract-function.tsx:193]: nit: reduce would be simpler here, so no need to filter out undefines

File: components/contract-functions/contract-function.tsx [hunk 2]
```diff
@@ -144,27 +152,78 @@ interface ContractFunctionsPanelProps {
...
+        if (filteredFunctions.length === 0) {
+          return undefined;
+        }
+        return {
+          extension: e.extension,
+          functions: filteredFunctions,
+        };
+      })
+      .filter((e) => e !== undefined) as ExtensionFunctions[];
+  }, [functionsWithExtension]);
```
Comment (assimovt) [inline components/contract-functions/contract-function.tsx:208]: nit: line breaks between functions. will add to lint rules

**Review by assimovt — APPROVED** [review]
lgtm

File: components/contract-functions/contract-function.tsx (hunk 2 above)
Comment (joaquim-verges) [reply components/contract-functions/contract-function.tsx:208]: prettier did this though

**Pull request merged**